extensions/amp-base-carousel/1.0/scroller.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as Preact from '../../../src/preact';
+import * as styles from './base-carousel.css';
+import {WithAmpContext} from '../../../src/preact/context';
+import {useEffect, useLayoutEffect, useRef} from '../../../src/preact';
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function Scroller(props) {
+  const {children, onSlideChange, currentSlide, setCurrentSlide} = props;
+  const slides = children || [];
```

Comment:
I think this should use `Preact.Children` util or something like that. It's not guaranteed that `children` are an array.

Reply:
IIUC in Preact [children are always an array](https://preactjs.com/guide/v8/differences-to-react/).

Reply:
I'm hearing that they are switching or planning to switch back to React's way since it started to create a lot of compatibility problems. Could you please try with a single child and let me know the state for today?

Reply:
But to confirm one thing: when AMP calls the `Carousel` component the `children` will always be an array, because that's how we implemented that. So, to try, we need to do something like `<MyComponent><div/></MyComponent>` and see what `MyComponent` gets as `children`. And one more nuance: we still need to contend with Preact/React compatibility a bit, since someone might want to use our component in React app. We can punt on this a bit, but let's see if we need to change it anyway for Preact.

Reply:
Good catch -- tried this out and got `TypeError: children.forEach is not a function`. I'll add and use `toChildArray`.